lib/exercism/xapi.rb

```diff
@@ -3,8 +3,12 @@ def self.exercises_api_url
     ENV.fetch('EXERCISES_API_URL') { "http://x.exercism.io" }
   end
 
-  def self.exists?(language, slug)
-    request('tracks', language, slug).status != 404
+  def self.exists?(language, slug=nil)
+    if slug
```

Comment:
You could say `request(*["tracks", language, slug].compact).status != 400` to remove the conditional, I think.